sklearn/datasets/_base.py

```diff
@@ -1381,14 +1381,37 @@ def _fetch_remote(remote, dirname=None):
     dirname : str
         Directory to save the file to.
 
+    progress : rich.progress.Progress
+
+    progress_task : rich.progress.Task
+    
     Returns
     -------
     file_path: str
         Full path of the created file.
     """
-
     file_path = remote.filename if dirname is None else join(dirname, remote.filename)
-    urlretrieve(remote.url, file_path)
+    if progress is not None:
```

Comment:
Currently, I am getting the following error:

```pytb
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[4], line 1
----> 1 fetch_lfw_people(show_progress=True)

File ~/Documents/packages/scikit-learn/sklearn/utils/_param_validation.py:214, in validate_params.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
    208 try:
    209     with config_context(
    210         skip_parameter_validation=(
    211             prefer_skip_nested_validation or global_skip_validation
    212         )
    213     ):
--> 214         return func(*args, **kwargs)
    215 except InvalidParameterError as e:
    216     # When the function is just a wrapper around an estimator, we allow
    217     # the function to delegate validation to the estimator, but we replace
    218     # the name of the estimator by the name of the function in the error
    219     # message to avoid confusion.
    220     msg = re.sub(
    221         r"parameter of \w+ must be",
    222         f"parameter of {func.__qualname__} must be",
    223         str(e),
    224     )

File ~/Documents/packages/scikit-learn/sklearn/datasets/_lfw.py:369, in fetch_lfw_people(data_home, funneled, resize, min_faces_per_person, color, slice_, download_if_missing, return_X_y, 
show_progress)
    261 @validate_params(
    262     {
    263         "data_home": [str, PathLike, None],
   (...)
    284     show_progress=False
    285 ):
    286     """Load the Labeled Faces in the Wild (LFW) people dataset \
    287 (classification).
    288 
   (...)
    367         .. versionadded:: 0.20
    368     """
--> 369     lfw_home, data_folder_path = _check_fetch_lfw(
    370         data_home=data_home, funneled=funneled, download_if_missing=download_if_missing, show_progress=show_progress
    371     )
    372     logger.debug("Loading LFW people faces from %s", lfw_home)
    374     # wrap the loader in a memoizing function that will return memmaped data
    375     # arrays for optimal memory usage

File ~/Documents/packages/scikit-learn/sklearn/datasets/_lfw.py:121, in _check_fetch_lfw(data_home, funneled, download_if_missing, show_progress)
    119     if progress:
    120         task = progress.add_task("[green]Downloading LFW data (~200MB)...", total=100)
--> 121     _fetch_remote(archive, dirname=lfw_home, progress=progress, progress_task=task)
    122 else:
    123     raise OSError("%s is missing" % archive_path)

File ~/Documents/packages/scikit-learn/sklearn/datasets/_base.py:1401, in _fetch_remote(remote, dirname, progress, progress_task)
   1399     is_new = True
   1400 else:
-> 1401     total = progress_task.total
   1402 def update_progress(count, block_size, total_size):
   1403     percentage = total * count * block_size / total_size

AttributeError: 'int' object has no attribute 'total'
```

It is important to have unit tests.
The challenge will be that we don't want to retrieve URL and we will need to mock this step and make sure that the progress bar works as if we were downloading.

Reply:
I fixed error

Reply:
Also, I think we can mocking downloading step using localhost server and mocking download with progress bar by downloading localhost files.